File: C/exercice3_4.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/types.h>
#include <sys/wait.h>
#include <time.h>
#include <errno.h>

int main()
{
	char* repertoire = malloc(50) ;
	char *pwd = getenv("PWD") ;
	pid_t fils ;
	while (1)
	{
		printf("Saisir un nom de repertoire:(taper quitter pour arrêter): ");
		scanf("%s", repertoire) ;
		if (strncmp(repertoire, "quitter", 7) == 0)
		{
			return (0) ;
		}
		fils = fork() ;
		if (fils == 0)
		{
			execlp("cp", "cp", "-r", pwd, repertoire, NULL) ;
		}
		else if (fils > 0)
		{
			wait(NULL) ;
		}
	}
	return 0;
}

File: Python/exercice3_4.py
import os

def main():
    pwd = os.getenv("PWD")
    
    while True:
        repertoire = input("Saisir un nom de répertoire (taper 'quitter' pour arrêter): ")
        
        if repertoire == "quitter":
            return 0

        fils = os.fork()

        if fils == 0:
            os.execlp("cp", "cp", "-r", pwd, repertoire, None)

        elif fils > 0:
            os.wait()
    
if __name__ == "__main__":
    main()